openwrt_api.py

```diff
@@ -0,0 +1,400 @@
+
+    ttl += 1
+    if curr_addr == dest_addr or ttl > max_hops:
+      break
+
+  return result
+
+def ping(dest_ip,count):
+  """
+  <Purpose>
+    Send ICMP ECHO_REQUEST to network hosts.
+
+  <Arguments>
+    dest_ip: the ip address to communicate with.
+    count: Stop after sending count ECHO_REQUEST packets.
+
+  <Exceptions>
+    RepyArgumentError when the host name is not valid types
+        or values
+
+  <Side Effects>
+    None
+
+  <Returns>
+    ping statistics as a dict, such as {'received': '2', 'jitter': '12.441', 
+    'avgping': '27.302', 'minping': '14.861', 'host': 'www.google.com', 
+    'maxping': '39.743', 'sent': '2'}
+    `host`: the target hostname that was pinged
+    `sent`: the number of ping request packets sent
+    `received`: the number of ping reply packets received
+    `minping`: the minimum (fastest) round trip ping request/reply
+        time in milliseconds
+    `avgping`: the average round trip ping time in milliseconds
+    `maxping`: the maximum (slowest) round trip ping time in
+        milliseconds
+    `jitter`: the standard deviation between round trip ping times
+        in milliseconds
+  """
+  if type(dest_ip) is not str:
+    raise RepyArgumentError("Provided host must be a string!")
+
+  if type(count) is not int:
+    raise RepyArgumentError("Provided count must be an int!")
+
+  # Check the input arguments (sanity)
+  if not emulcomm._is_valid_ip_address(dest_ip):
+    raise RepyArgumentError("Provided dest_ip is not valid! IP: '" + dest_ip + "'")
+
+  if count < 1:
+    raise RepyArgumentError("Provided count must be more than 0")
+
+  process = portable_popen.Popen(['ping', '-c', str(count), dest_ip],)
+  ping_output, err = process.communicate()
+  return _parse(str(ping_output))
+
+def get_station(interface):
+  """
+   <Purpose>
+      To get station statistic information such as the amount of tx/rx bytes, the 
+      last TX bitrate (including MCS rate) you can do
+
+   <Arguments>
+      interface: the name of the interface to gather network information.
+
+   <Exceptions>
+      RepyArgumentError is raised if the argument isn't valid types or values
+
+   <Side Effects>
+      None
+
+   <Returns>
+     station statistic informatio as a dict such as {{'signal': '-77 [-78, -83] dBm', 
+    'station': '54:4e:90:06:f3:6f', 'tx bitrate': '57.8 MBit/s MCS 5 short GI', 'rx 
+    bitrate': '1.0 MBit/s', 'signal_avg': '-88 [-90, -92] dBm'}, {'signal': '-40 [-42, 
+    -44] dBm', 'station': 'ac:81:12:53:8e:0f', 'tx bitrate': '65.0 MBit/s MCS 6 short 
+    GI', 'rx bitrate': '18.0 MBit/s', 'signal_avg': '-42 [-44, -48] dBm'}}
+
+  """
+  if type(interface) is not str:
+    raise RepyArgumentError("get_station() takes a string as argument.")
+
+  # Check the input arguments (sanity)
+  if interface not in get_network_interface():
+    raise RepyArgumentError("interface "+ interface + " is not available.")
+
+  iw_process = portable_popen.Popen(['iw', 'dev',interface, 'station','dump'])
```

Comment:
Is `iw` guaranteed to exist? E.g., a router that has no wireless hardware/drivers (and thus no need for the WiFi tools).